### PR TITLE
[build] `make_lazy_extractors`: Force running without plugins

### LINKED ISSUE #7859: [Compile] Error in `lazy_extractors.py`
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2023.07.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

System: `Linux Mint 20.3 64bit`
GNU Make 4.2.1
yt-dlp version `2023.08.16.1 (stable), head=876b70c8edf4c0147f180bd981fbc4d625cbfb9c`

Recently, since commit `378ae9f`, compiling itself proceeds as always without error, but executing `yt-dlp "URL"` gives:

```shell
./yt-dlp https://www.youtube.com/watch?v=fH81o3yJ5yU
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "./yt-dlp/__main__.py", line 17, in <module>
  File "./yt-dlp/yt_dlp/__init__.py", line 1008, in main
  File "./yt-dlp/yt_dlp/__init__.py", line 998, in _real_main
  File "./yt-dlp/yt_dlp/YoutubeDL.py", line 3518, in download
  File "./yt-dlp/yt_dlp/YoutubeDL.py", line 3493, in wrapper
  File "./yt-dlp/yt_dlp/YoutubeDL.py", line 1550, in extract_info
  File "./yt-dlp/yt_dlp/extractor/lazy_extractors.py", line 187, in suitable
NameError: name 'YoutubeIE' is not defined
```
[lazy_extractors.tar.gz](https://github.com/yt-dlp/yt-dlp/files/12356907/lazy_extractors.tar.gz)

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
./yt-dlp --ignore-config -vU
[debug] Command-line config: ['--ignore-config', '-vU']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.08.16.1 [876b70c8e] (zip)
[debug] Python 3.8.10 (CPython x86_64 64bit) - Linux-5.4.0-156-generic-x86_64-with-glibc2.29 (OpenSSL 1.1.1f  31 Mar 2020, glibc 2.31)
[debug] exe versions: ffmpeg N-111711-gd295b6b693-Nico-20230808 (fdk,setts), ffprobe N-111711-gd295b6b693-Nico-20230808, phantomjs 2.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.18.0, brotli-1.0.9, certifi-2023.07.22, mutagen-1.46.0, secretstorage-3.3.3, sqlite3-2.6.0, websockets-11.0.3, xattr-0.9.6
[debug] Proxy map: {}
[debug] Loaded 1862 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.08.16.1
Current Build Hash: 07b5f41679cb34e353f7289a8c3ee26c1b379a351508aba61eb9ac7f903d0982
yt-dlp is up to date (stable@2023.08.16.1)
```

### PR DESCRIPTION
**Pull request by Grub4K**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Fixes #7859

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [x] Core bug fix/improvement

</details>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1897,6 +1897,7 @@ In other words, the file structure on the disk looks something like:
                 myplugin.py
 
 yt-dlp looks for these `yt_dlp_plugins` namespace folders in many locations (see below) and loads in plugins from **all** of them.
+Set the environment variable `YTDLP_NO_PLUGINS` to something nonempty to disable loadig plugins entirely.
```
Comment (bashonly) [inline README.md:1900]:
```suggestion
Set the environment variable `YTDLP_NO_PLUGINS` to something nonempty to disable loading plugins entirely.
```

File: devscripts/make_lazy_extractors.py [hunk 2]
```diff
@@ -34,18 +33,14 @@ class {name}({bases}):
 
 
 def main():
-    lazy_extractors_filename = get_filename_args(default_outfile='yt_dlp/extractor/lazy_extractors.py')
-    if os.path.exists(lazy_extractors_filename):
-        os.remove(lazy_extractors_filename)
+    os.environ['YTDLP_NO_PLUGINS'] = 'true'
+    os.environ['YTDLP_NO_LAZY_EXTRACTORS'] = 'true'
```
Comment (bashonly) [inline devscripts/make_lazy_extractors.py:37]: `'1'` would match our documentation for `YTDLP_NO_LAZY_EXTRACTORS` but it doesn't really matter
```suggestion
    os.environ['YTDLP_NO_PLUGINS'] = '1'
    os.environ['YTDLP_NO_LAZY_EXTRACTORS'] = '1'
```

**Review by bashonly — APPROVED** [review]

File: devscripts/make_lazy_extractors.py (hunk 2 above)
Comment (Grub4K) [reply devscripts/make_lazy_extractors.py:37]: I'd rather change the documentation to reflect the implementation, so just say `nonempty` for `YTDLP_NO_LAZY_EXTRACTORS`

**Review by bashonly — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1897,6 +1897,7 @@ In other words, the file structure on the disk looks something like:
                 myplugin.py
 
 yt-dlp looks for these `yt_dlp_plugins` namespace folders in many locations (see below) and loads in plugins from **all** of them.
+Set the environment variable `YTDLP_NO_PLUGINS` to something nonempty to disable loading plugins entirely.
```
Comment (pukkandan) [inline README.md:1900]: This should not be user-facing and should be folded into something like `--plugin-dir none` imo

File: README.md [hunk 4]
```diff
@@ -278,7 +278,7 @@ py -m bundle.py2exe
 * **`devscripts/update-version.py`** - Update the version number based on the current date.
 * **`devscripts/set-variant.py`** - Set the build variant of the executable.
 * **`devscripts/make_changelog.py`** - Create a markdown changelog using short commit messages and update `CONTRIBUTORS` file.
-* **`devscripts/make_lazy_extractors.py`** - Create lazy extractors. Running this before building the binaries (any variant) will improve their startup performance. Set the environment variable `YTDLP_NO_LAZY_EXTRACTORS=1` if you wish to forcefully disable lazy extractor loading.
+* **`devscripts/make_lazy_extractors.py`** - Create lazy extractors. Running this before building the binaries (any variant) will improve their startup performance. Set the environment variable `YTDLP_NO_LAZY_EXTRACTORS` to something nonempty to forcefully disable lazy extractor loading.
```
Comment (pukkandan) [inline README.md:281]: I preferred the older doc. User doesn't have to know whether other values work or not.

File: README.md (hunk 3 above)
Comment (coletdjnz) [reply README.md:1900]: I agree, I didn't realise this was merged. It should not be documented..

 --no-plugins is being added in https://github.com/yt-dlp/yt-dlp/pull/11305

Comment (Grub4K) [reply README.md:1900]: I disagree. This is a valid usecase and `--no-plugins` was not yet there. It doesn't harm to have a variable like this, which is also namespaced as to not clobber other variables names.

**Pull request merged**